client/gutenberg/editor/components/header/opt-out-menu-item/index.jsx

```diff
@@ -0,0 +1,67 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * WordPress Dependencies
+ */
+import { MenuItem } from '@wordpress/components';
+import {
+	composeAnalytics,
+	recordGoogleEvent,
+	recordTracksEvent,
+	withAnalytics,
+	bumpStat,
+} from 'state/analytics/actions';
+import { setSelectedEditor } from 'state/selected-editor/actions';
+import getCurrentRoute from 'state/selectors/get-current-route';
+import { navigate } from 'state/ui/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export const OptOutMenuItem = ( {
+	classicEditorRoute,
+	navigate: redirect,
+	optOut,
+	siteId,
+	translate,
+} ) => {
+	const switchToClassicEditor = () => {
+		optOut( siteId );
+		redirect( classicEditorRoute );
```

Comment:
Since this doesn't wait for the opt-out API request to finish, it might be safer to add the `?force=true` param, so Calypso can bypass the selected editor check, just in case the selected editor is not updated yet during the redirection (we do this in Calypsoify: D20254-code)